packages/features/calendars/lib/getTodaysDateInTimeZone.ts

```diff
@@ -0,0 +1,25 @@
+import timezone from "dayjs/plugin/timezone";
+import utc from "dayjs/plugin/utc";
+
+import dayjs from "@calcom/dayjs";
+
+// with new Date(), we get browser's current date.
+// The below function gets today's date w.r.t to selected timezone.
+// Example :
+// Consider browser current date-time is - "Sat Jul 20 2024 01:00:00 AM (Asia/Kolkata GMT +5:30)".
+// And, if preferred timezone is "Pacific/Pago GMT -11:00", we need to get today's date as "Fri Jul 19".
+export function getTodaysDateInTimeZone(selectedTimeZone: string | undefined): Date {
+  // These plugins is requred for dayjs to calculate today's date w.r.t input timezone
+  dayjs.extend(utc);
+  dayjs.extend(timezone);
+
+  const todayDateInPreferredTZ = dayjs().tz(selectedTimeZone);
+
+  // 'todayDateInPreferredTZ' is dayjs date, we need to convert this into type 'Date',
+  // as input required for getAvailableDatesInMonth() is 'Date' type.
+  const offsetMinutes = dayjs().utcOffset();
+  const jsDate = new Date(`${todayDateInPreferredTZ.format("YYYY-MM-DDTHH:mm:ss")}Z`); // UTC format
```

Comment:
Instead of `jsDate` can we set `localDate` or only `date` ?

Reply:
i am intending to be Date of type Date (JS/TS native object), as we can also get date from Dayjs().toDate() but this will again give date in local timezone